### PR TITLE
Update Captions based on new sdk changes and add changeCaptionLanguage 

### PR DESCRIPTION
**Pull request by carocao-msft**
# What
Updated events, functions and variable names based on calling sdk changes
Cleaned up duplicate captions event registrations 
Clear captions content when turn captions off 
Use global spoken language state to determine show/hide captions modal
Enable new functionality change caption language 

# Why

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

**Comment by edwardlee-msft** [comment]
Can we keep this as a Draft PR until the tgz file is replaced with a working beta/public version?

File: packages/calling-stateful-client/src/CallSubscriber.ts [hunk 1]
```diff
@@ -211,13 +209,15 @@ export class CallSubscriber {
...
     // subscribe to captions here so that we don't call captions when call is not initialized
-    if (_isTeamsMeetingCall(this._call) && this._call.state === 'Connected' && !this._captionsSubscriber) {
-      this._captionsSubscriber = new CaptionsSubscriber(
-        this._callIdRef,
-        this._context,
-        this._call.feature(Features.TeamsCaptions)
-      );
-      this._call.off('stateChanged', this.initCaptionSubscriber);
+    if (this._call.state === 'Connected' && !this._captionsSubscriber) {
+      if (this._call.feature(Features.Captions).captions.captionsType === 'TeamsCaptions') {
```
Comment (edwardlee-msft) [inline packages/calling-stateful-client/src/CallSubscriber.ts:213]: captionsType changed to kind

File: packages/calling-component-bindings/src/handlers/createCommonHandlers.ts [hunk 2]
```diff
@@ -26,8 +26,10 @@ import { disposeAllLocalPreviewViews, _isInCall, _isInLobbyOrConnecting, _isPrev
 import { CommunicationUserIdentifier, PhoneNumberIdentifier, UnknownIdentifier } from '@azure/communication-common';
 /* @conditional-compile-remove(PSTN-calls) */
 import { CommunicationIdentifier } from '@azure/communication-common';
-/* @conditional-compile-remove(video-background-effects) */
+/* @conditional-compile-remove(video-background-effects) */ /* @conditional-compile-remove(close-captions) */
```
[the remark this answers is not in the document]
Comment (carocao-msft) [reply packages/calling-component-bindings/src/handlers/createCommonHandlers.ts:29]: It's the conditional compile line, going to be removed when we ga

File: packages/calling-component-bindings/src/handlers/createCommonHandlers.ts [hunk 3]
```diff
@@ -507,21 +509,26 @@ export const createDefaultCommonCallingHandlers = memoizeOne(
...
     const onStopCaptions = async (): Promise<void> => {
-      await call?.feature(Features.TeamsCaptions).stopCaptions();
+      const captionsFeature = call?.feature(Features.Captions) as unknown as TeamsCaptions;
+      await captionsFeature.stopCaptions();
     };
     /* @conditional-compile-remove(close-captions) */
     const onSetSpokenLanguage = async (language: string): Promise<void> => {
-      await call?.feature(Features.TeamsCaptions).setSpokenLanguage(language);
+      const captionsFeature = call?.feature(Features.Captions) as unknown as TeamsCaptions;
+      await captionsFeature.setSpokenLanguage(language);
```
[the remark this answers is not in the document]
Comment (carocao-msft) [reply packages/calling-component-bindings/src/handlers/createCommonHandlers.ts]: Updated

File: packages/calling-stateful-client/src/CallSubscriber.ts (hunk 1 above)
Comment (carocao-msft) [reply packages/calling-stateful-client/src/CallSubscriber.ts:213]: Updated in another PR

File: packages/calling-stateful-client/src/CallContext.ts [hunk 4]
```diff
@@ -789,6 +789,17 @@ export class CallContext {
       }
     });
   }
+
+  /* @conditional-compile-remove(close-captions) */
+  public clearCaption(callId: string): void {
```
Comment (JamesBurnside) [inline packages/calling-stateful-client/src/CallContext.ts:794]: nit: `clearCaptions` (plural)

File: packages/calling-stateful-client/src/CallSubscriber.ts [hunk 5]
```diff
@@ -211,13 +209,15 @@ export class CallSubscriber {
...
   private initCaptionSubscriber = (): void => {
     // subscribe to captions here so that we don't call captions when call is not initialized
-    if (_isTeamsMeetingCall(this._call) && this._call.state === 'Connected' && !this._captionsSubscriber) {
-      this._captionsSubscriber = new CaptionsSubscriber(
-        this._callIdRef,
-        this._context,
-        this._call.feature(Features.TeamsCaptions)
-      );
-      this._call.off('stateChanged', this.initCaptionSubscriber);
+    if (this._call.state === 'Connected' && !this._captionsSubscriber) {
```
Comment (JamesBurnside) [inline packages/calling-stateful-client/src/CallSubscriber.ts:212]: Having to read callState to understand if we can have access to a captions feature isn't great :/

@edwardlee-msft is this the documented way developers have to access captions - they must wait for the callstate to be connected _then_ access the captions API?

Ideally as a developer if I have a call object, regardless of the call state I could call `call.feature(Features.Captions).caption` and set up my subscriptions

File: packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts [hunk 6]
```diff
@@ -997,20 +1002,23 @@ export class AzureCommunicationCallAdapter<AgentType extends CallAgent | BetaTea
...
   }
 
   /* @conditional-compile-remove(close-captions) */
   private unsubscribeFromCaptionEvents(): void {
-    if (this.call && this.call.state === 'Connected' && _isTeamsMeetingCall(this.call)) {
-      this._call?.feature(Features.TeamsCaptions).off('captionsReceived', this.captionsReceived.bind(this));
-      this._call
-        ?.feature(Features.TeamsCaptions)
-        .off('isCaptionsActiveChanged', this.isCaptionsActiveChanged.bind(this));
+    if (this.call && this.call.state === 'Connected' && this.context.getState().isTeamsCall) {
```
Comment (JamesBurnside) [inline packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts:1016]: Is it intentionally we only ever unsubscribe from captions when callstate is connected? What if the call ends and then I call unsubscribe?

**Review by JamesBurnside — APPROVED** [review]
Some small comments are subscriptions but LGTM!

File: packages/calling-stateful-client/src/CallSubscriber.ts (hunk 5 above)
Comment (carocao-msft) [reply packages/calling-stateful-client/src/CallSubscriber.ts:212]: Yeah we can only subscribe to captions when the call is connected

File: packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts (hunk 6 above)
Comment (carocao-msft) [reply packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts:1016]: The captions API can only be accessed when the call is connected. So this is intentional.

**Review by prabhjot-msft — APPROVED** [review]

**Pull request merged**